src/components/user-settings.tsx

```diff
@@ -162,11 +162,27 @@ export default function UserSettings() {
 	} );
 
 	const onRemoveSnapshots = useCallback( async () => {
-		if ( snapshots?.length === 0 ) {
+		if ( ! allSnapshots || allSnapshots.length === 0 ) {
 			return;
 		}
-		await deleteAllSnapshots( snapshots );
-	}, [ snapshots, deleteAllSnapshots ] );
+
+		const CANCEL_BUTTON_INDEX = 0;
+		const DELETE_BUTTON_INDEX = 1;
+
+		const { response } = await getIpcApi().showMessageBox( {
+			type: 'warning',
+			message: __( 'Delete all demo sites' ),
+			detail: __(
+				'All demo sites databases that exist for your WordPress.com account, along with all posts, pages, comments, and media, will be lost.'
```

Comment:
Should we drop the `databases` word to make it clearer that demo sites are deleted, and not only databases?

```suggestion
				'All demo sites that exist for your WordPress.com account, along with all posts, pages, comments, and media, will be lost.'
```